x-pack/platform/plugins/shared/ml/public/application/jobs/jobs_list/components/datafeed_chart_flyout/datafeed_chart_flyout.tsx

```diff
@@ -421,10 +423,12 @@ export const DatafeedChartFlyout: FC<DatafeedChartFlyoutProps> = ({
                             )
                               return;
 
-                            onModelSnapshotAnnotationClick(
-                              // @ts-expect-error property 'modelSnapshot' does not exist on type
-                              annotations.lines[0].datum.modelSnapshot
-                            );
+                            if (onModelSnapshotAnnotationClick) {
+                              onModelSnapshotAnnotationClick(
+                                // @ts-expect-error property 'modelSnapshot' does not exist on type
+                                annotations.lines[0].datum.modelSnapshot
```

Comment:
Will this access path `annotations.lines[0].datum.modelSnapshot` always be populated as shown? Do we need to add anything to handle one of these not being defined?